src/account/PluginManagerInternals.sol

```diff
@@ -138,32 +134,21 @@ abstract contract PluginManagerInternals is IPluginManager {
         );
     }
 
-    function _installPlugin(address plugin, bytes32 manifestHash, bytes memory pluginInstallData) internal {
+    function _installPlugin(address plugin, PluginManifest calldata manifest, bytes memory pluginInstallData)
+        internal
+    {
         AccountStorage storage _storage = getAccountStorage();
 
         if (plugin == address(0)) {
             revert NullPlugin();
         }
 
-        // Check if the plugin exists.
-        if (_storage.pluginManifestHashes.contains(plugin)) {
-            revert PluginAlreadyInstalled(plugin);
-        }
-
+        // TODO: do we need this check? Or switch to a non-165 checking function?
         // Check that the plugin supports the IPlugin interface.
```

Comment:
i like the check as a safety mechanism, because installation of non plugin contracts can be dangerous (installing ERC20 tokens/ERC20.transfer.selector as an execution selector to transfer contract tokens out)

Are you thinking of moving it to the SDK to reduce pollution in the tx traces? That's something I'll support. I think we can say with high confidence that most accounts are used with their paired SDKs today, and will be the case in the future, so safety checks in the SDK are equally effective as on chain.